panda/src/text/textAssembler.cxx

```diff
@@ -1781,6 +1781,76 @@ shape_buffer(hb_buffer_t *buf, PlacedGlyphs &placed_glyphs, PN_stdfloat &xpos,
              const TextProperties &properties) {
 
 #ifdef HAVE_HARFBUZZ
+  // before we do *anything*, check to see if small_caps == true, and change the characters before
+  // setting up everything else; check to see if small_caps == true, and change "glyph_index" if so
+  // (glyph_index *should* be the character, since we've not told harfbuff what to do with it yet)
+  // but first, we need some of these defined sooner rather than later:
+  DynamicTextFont *font = DCAST(DynamicTextFont, properties.get_font());
+
+  if (properties.has_small_caps()) {
```

Comment:
This checks whether the property has been set, not whether it's actually set to true (use `get_small_caps`).